packages/block-editor/src/components/block-popover/index.js

```diff
@@ -77,30 +77,36 @@ function BlockPopover(
 		};
 	}, [ selectedElement ] );
 
-	const { isZoomOut, sectionRootClientId, getParentSectionBlock } = useSelect(
-		( select ) => {
-			const {
-				isZoomOut: isZoomOutSelector,
-				getSectionRootClientId,
-				getParentSectionBlock: getParentSectionBlockFn,
-			} = unlock( select( blockEditorStore ) );
-
-			const root = getSectionRootClientId();
-			return {
-				sectionRootClientId: root,
-				isZoomOut: isZoomOutSelector(),
-				getParentSectionBlock: getParentSectionBlockFn,
-			};
-		},
-		[]
-	);
+	const {
+		isZoomOut,
+		sectionRootClientId,
+		getParentSectionBlock,
+		getBlockOrder,
+	} = useSelect( ( select ) => {
+		const {
+			isZoomOut: isZoomOutSelector,
+			getSectionRootClientId,
+			getParentSectionBlock: getParentSectionBlockFn,
+			getBlockOrder: getBlockOrderFn,
+		} = unlock( select( blockEditorStore ) );
+
+		const root = getSectionRootClientId();
+		return {
+			sectionRootClientId: root,
+			isZoomOut: isZoomOutSelector(),
+			getParentSectionBlock: getParentSectionBlockFn,
+			getBlockOrder: getBlockOrderFn,
+		};
+	}, [] );
 
 	// These elements are used to position the zoom out view vertical toolbar
 	// correctly, relative to the selected section.
 	const rootSectionElement = useBlockElement( sectionRootClientId );
 	const parentSectionElement = useBlockElement(
 		getParentSectionBlock( clientId ) ?? clientId
 	);
+	const isSectionSelected =
+		getBlockOrder( sectionRootClientId ).includes( clientId );
```

Comment:
Aside: I'm starting to see this pattern a fair bit. Wondering if it's time for a private selector.

Reply:
This is slightly different. Riad added section selectors but he's counting on sections always being content only containers which is not true yet.

Reply:
I see 👍